File: 0875-koko-eating-bananas/0875-koko-eating-bananas.py
class Solution(object):
    def calHours(self,piles,mid):
        hours=0
        for i in range(0,len(piles)):
            hours+=(piles[i]+mid-1)//mid
        return hours
    def minEatingSpeed(self, piles, h):    
        low,high=1,max(piles)
        ans=high
        while(low<=high):
            mid=(low+high)//2
            hours=self.calHours(piles,mid)
            if hours<=h:
                ans=mid
                high=mid-1
            else:
                low=mid+1
        return ans

File: 0875-koko-eating-bananas/0875-koko-eating-bananas.cpp
class Solution {
public:
    long long calculateHours(vector<int>& piles,int k){
        long long hours=0;
        for(int i=0;i<piles.size();i++){
            hours+=ceil((double)piles[i]/k);
        }
        return hours;
    }
    int minEatingSpeed(vector<int>& piles, int h) {
        int st=1,end =* max_element(piles.begin(),piles.end());
        int ans;
        while(st<=end){
            int mid=st+(end-st)/2;
            long long reqHours=calculateHours(piles,mid);
            if(reqHours<=h){
                ans=mid;
                end=mid-1;
            }else{
                st=mid+1;
            }
        }

        return ans;
    }
};